structs/schedule.go

```diff
@@ -0,0 +1,248 @@
+package structs
+
+// TODO(CaptainHayashi): this probably doesn't belong in structs.
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/UniversityRadioYork/myradio-go"
+)
+
+// ScheduleItem is an interface containing information about one item in a URY schedule.
+type ScheduleItem interface {
+	// GetName gets the display name of the schedule item.
+	// The website page context is passed to resolve, for example, the sustainer name.
+	GetName(context *PageContext) string
+
+	// GetDesc gets the description of the schedule item.
+	// The website page context is passed to resolve, for example, the
+	// sustainer description.
+	GetDesc(context *PageContext) string
+
+	// GetStart gets the start time of the schedule item.
+	GetStart() time.Time
+
+	// GetFinish gets the finish time of the schedule item.
+	GetFinish() time.Time
+
+	// GetBlock gets the block name of the schedule item.
+	GetBlock() string
+
+	// IsSustainer gets whether this schedule item is the URY sustainer.
+	IsSustainer() bool
+
+	// HasPage is true when this schedule item has an associated page.
+	HasPage() bool
+
+	// GetPageURL gets the root-relative URL to this schedule item's page.
+	GetPageURL() string
```

Comment:
Maybe this should return `(string, bool)` instead of `HasPage` being a separate thing.  Or even `(URL, bool)`